cli-plugins/apps/debug-workload.hbs.md

```diff
@@ -67,6 +67,72 @@ There are known errors that will make the workload enter in an error or unknown
+
+## <a id="steps-failure"></a> Supply Chain steps
+
+After a workload is created via `tanzu apps workload create (or) apply`, the supply chain steps that it goes through, and the current condition of each, can be reviewed in the output from `tanzu apps workload get`.
+
+The output looks like the following:
+
+```bash
+...
+📦 Supply Chain
+   name:   source-to-url
+
+   NAME               READY   HEALTHY   UPDATED   RESOURCE
+   source-provider    True    True      71m       gitrepositories.source.toolkit.fluxcd.io/spring-petclinic
+   image-provider     True    True      70m       images.kpack.io/spring-petclinic
+   config-provider    True    True      69m       podintents.conventions.carto.run/spring-petclinic
+   app-config         True    True      69m       configmaps/spring-petclinic
+   service-bindings   True    True      69m       configmaps/spring-petclinic-with-claims
+   api-descriptors    True    True      69m       configmaps/spring-petclinic-with-api-descriptors
+   config-writer      True    True      69m       runnables.carto.run/spring-petclinic-config-writer
+
+🚚 Delivery
+   name:   delivery-basic
+
+   NAME              READY   HEALTHY   UPDATED   RESOURCE
+   source-provider   True    True      69m       imagerepositories.source.apps.tanzu.vmware.com/spring-petclinic-delivery
+   deployer          True    True      69m       apps.kappctrl.k14s.io/spring-petclinic
+
+💬 Messages
+   No messages found.
+...
+```
+
+In the `Supply Chain` section, the supply chain steps associated with the workload are displayed in a table, alongside their reconcile state. In the event there's a failure for a given step, the `READY` column value will be `Unknown` or `False`, and the `HEALTHY` column value will be `False`.
+
+In case there is a resource in `Unknown` or `False` status, it can be inspected with:
+
+```bash
+kubectl describe <resource-name>
+```
+
+Where `resource-name` refers to the name of the stamped out resource, displayed in `RESOURCE` column.
+
+For example, if something like this is retrieved in `tanzu apps workload get` command:
+
+```bash
+NAME               READY   HEALTHY   UPDATED   RESOURCE
+source-provider    False   False     3h12m     gitrepositories.source.toolkit.fluxcd.io/spring-petclinic
+```
+
+Whatever is going on with this resource can be checked with:
+
+```bash
+kubectl describe gitrepositories.source.toolkit.fluxcd.io/spring-petclinic
+```
+
+In addition, the `Messages` section of the `tanzu apps workload get` command output notifies could give a hint as to what went wrong in the process.
+
+For example, a message like this could be shown:
+
+```bash
+💬 Messages
+   Workload [HealthyConditionRule]:   failed to checkout and determine revision: failed to resolve commit object for '425ae9a2a2f84d195a9f3862668e8b2abf81418a': object not found
+```
+
+This could mean that the given commit does not belong to the specified branch or does not exist in the repo.
```

Comment:
This could mean that the given commit does not belong to the specified branch or does not exist in the repo.

## <a id="additional-troubleshooting"></a> Additional Troubleshooting References
Additional Workload troubleshooting tips can be found on the [Troubleshoot using Tanzu Application Platform page](../../troubleshooting-tap/troubleshoot-using-tap.hbs.md).